### PR TITLE
Feat/tunnel

### PR DESCRIPTION
**Pull request by wussler**
New functions for ECDHE exchange

### CODE REVIEW

File: lib/key/utils.js [hunk 1]
```diff
@@ -101,3 +104,29 @@ export async function cloneKey(inputKey) {
...
+    return openpgp.crypto.publicKey.elliptic.ecdh.genPublicEphemeralKey(
+        oid,
+        curveObj.cipher,
+        curveObj.hash,
+        Q,
+        fingerprint
+    );
+}
+
+export async function genPrivateEphemeralKey(curve, V, d, fingerprint) {
```
Comment (dhoko) [inline lib/key/utils.js]: Can you add the JSDoc ?
Instead of having 4 args, can you use destructuring ?

File: lib/key/utils.js [hunk 2]
```diff
@@ -101,3 +104,29 @@ export async function cloneKey(inputKey) {
     const [key] = await getKeys(inputKey.toPacketlist().write());
     return key;
 }
+
+// ECDHE
+export async function genPublicEphemeralKey(curve, Q, fingerprint) {
```
Comment (dhoko) [inline lib/key/utils.js]: Can you add the JSDoc ?

File: lib/key/utils.js [hunk 3]
```diff
@@ -2,11 +2,14 @@ import { openpgp } from '../openpgp';
...
 // returns promise for generated RSA public and encrypted private keys
-export const generateKey = (options) => {
+export async function generateKey(options) {
     options.date = typeof options.date === 'undefined' ? serverTime() : options.date;
     return openpgp.generateKey(options);
-};
-export const generateSessionKey = (algorithm) => openpgp.crypto.generateSessionKey(algorithm);
+}
+
+export async function generateSessionKey(algorithm) {
```
Comment (dhoko) [inline lib/key/utils.js]: `export const generateSessionKey = openpgp.crypto.generateSessionKey`
No need to wrap it again ;) and it's already async

Comment (wussler) [reply lib/key/utils.js]: Thanks, I'll remove from the commit

**Pull request closed without merging**